crates/cli-support/src/js/mod.rs

```diff
@@ -2634,63 +2635,115 @@ impl<'a, 'b> SubContext<'a, 'b> {
+            c.contents.push_str(fn_name);
+            c.contents.push_str(fn_args);
+            c.contents.push_str(";\n}\n");
+        };
+        let finish_j2r = |mut j2r: Js2Rust| -> Result<(_, _, _), Error> {
+            Ok(j2r
+                .process(descriptor.unwrap_function(), &export.function.arg_names)?
+                .finish(
+                    "",
+                    &format!("wasm.{}", wasm_name),
+                    ExportedShim::Named(&wasm_name),
+                ))
+        };
+        match &export.method_kind {
+            decode::MethodKind::Constructor => {
+                fn_name = "constructor";
+                let mut j2r = Js2Rust::new(fn_name, self.cx);
+                j2r.constructor(Some(class_name));
+                let (js, ts, raw_docs) = finish_j2r(j2r)?;
+                let class = require_class(&mut self.cx.exported_classes, class_name);
+                if class.has_constructor {
+                    bail!("found duplicate constructor `{}`", export.function.name);
+                }
+                class.has_constructor = true;
+                let docs = docs(raw_docs);
+                method(class, &docs, fn_name, "", &js, &ts);
+                Ok(())
+            }
+            decode::MethodKind::Operation(operation) => {
+                let mut j2r = Js2Rust::new(fn_name, self.cx);
+                let mut static_prfx = "";
+                if operation.is_static {
+                    static_prfx = "static ";
+                } else {
+                    j2r.method(export.consumed);
+                }
+                let (js, ts, raw_docs) = finish_j2r(j2r)?;
+                let class = require_class(&mut self.cx.exported_classes, class_name);
+                let docs = docs(raw_docs);
+                match operation.kind {
+                    decode::OperationKind::Getter(maybe_getter_name) => {
+                        if export.function.renamed_via_js_name {
+                            method(class, &docs, fn_name, "", &js, &ts);
+                            getter_from_method_call(class, &docs, fn_name, maybe_getter_name);
+                        } else {
+                            let getter_name = if maybe_getter_name.is_empty() {
+                                fn_name
+                            } else {
+                                maybe_getter_name
+                            };
+                            method(class, &docs, getter_name, "get ", &js, &ts);
+                        }
+                    }
+                    decode::OperationKind::Setter(maybe_setter_name) => {
+                        if export.function.renamed_via_js_name {
+                            method(class, &docs, fn_name, "", &js, &ts);
+                            setter_from_method_call(class, &docs, fn_name, &js, maybe_setter_name);
+                        } else {
+                            let getter_name = if maybe_setter_name.is_empty() {
```

Comment:
Should `maybe_setter_name` be an `Option` instead of a `&str` if we test if it's empty here?

Reply:
Or actually better yet, never have it as an empty string but the backend codegen does the `unwrap_or` for us ahead of time

Reply:
Hummm... There must be a way to known if the user provided a name for the getter/setter attribute and `Option` would be more intuitive